packages/dev/core/src/Misc/screenshotTools.ts

```diff
@@ -95,32 +77,57 @@ export function CreateScreenshot(
             undefined,
             quality
         );
-    } else {
-        engine.onEndFrameObservable.addOnce(() => {
-            const renderingCanvas = engine.getRenderingCanvas();
-            if (renderContext && renderingCanvas) {
-                renderContext.drawImage(renderingCanvas, offsetX, offsetY, newWidth, newHeight);
-            }
+        return;
+    }
 
-            if (screenshotCanvas) {
-                if (forceDownload) {
-                    Tools.EncodeScreenshotCanvasData(screenshotCanvas, undefined, mimeType, undefined, quality);
-                    if (successCallback) {
-                        successCallback("");
-                    }
-                } else {
-                    Tools.EncodeScreenshotCanvasData(screenshotCanvas, successCallback, mimeType, undefined, quality);
-                }
+    engine.onEndFrameObservable.addOnce(() => {
+        if (!screenshotCanvas) {
+            screenshotCanvas = document.createElement("canvas");
+        }
+        screenshotCanvas.width = width;
+        screenshotCanvas.height = height;
+
+        const renderContext = screenshotCanvas.getContext("2d");
+        const renderingCanvas = engine.getRenderingCanvas();
+        if (!renderContext || !renderingCanvas) {
+            Logger.Error("Failed to create screenshot. Rendering context or rendering canvas is not available.");
+            return;
+        }
+
+        const srcWidth = renderingCanvas.width;
+        const srcHeight = renderingCanvas.height;
+        const destWidth = screenshotCanvas.width;
+        const destHeight = screenshotCanvas.height;
+
+        // Calculate scale factors for width and height. Use the larger of the two to fill the screenshot.

```

Comment:
I think the second part of the comment is not accurate, as it depends on the `useFill` parameter value?

Reply:
Oops, you're right. Updated.